sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -87,7 +87,7 @@ private[sql] class CacheManager extends Logging {
       query: Dataset[_],
       tableName: Option[String] = None,
       storageLevel: StorageLevel = MEMORY_AND_DISK): Unit = writeLock {
-    val planToCache = query.queryExecution.analyzed
+    val planToCache = query.queryExecution.analyzed.canonicalized
```

Comment:
do we still need these changes? `LogicalPlan.canonicalized` is a lazy val and we don't have performance penalty even we use un-canonicalized plan as key.